### PR TITLE
Protect jsonfilelog writes with mutex

### PR DESCRIPTION
**Pull request by LK4D4**
Fixes panic on jenkins.
ping @estesp

### CODE REVIEW

File: daemon/logger/jsonfilelog/jsonfilelog.go [hunk 1]
```diff
@@ -29,12 +31,18 @@ func New(filename string) (logger.Logger, error) {
 
 // Log converts logger.Message to jsonlog.JSONLog and serializes it to file
 func (l *JSONFileLogger) Log(msg *logger.Message) error {
+	l.mu.Lock()
 	err := (&jsonlog.JSONLog{Log: string(msg.Line) + "\n", Stream: msg.Source, Created: msg.Timestamp}).MarshalJSONBuf(l.buf)
 	if err != nil {
 		return err
```
Comment (estesp) [inline daemon/logger/jsonfilelog/jsonfilelog.go]: unlock here or add `defer l.mu.Unlock()`?

Comment (LK4D4) [inline daemon/logger/jsonfilelog/jsonfilelog.go]: Indeed, thanks!

**Comment by LK4D4** [comment]
@estesp Fixed

**Comment by jfrazelle** [comment]
wooohoooo

**Comment by jfrazelle** [comment]
jenkins failed bc of TestPullVerified so unrelated, LGTM

**Comment by estesp** [comment]
LGTM

**Pull request merged**